### ISSUE TITLE
Add thorough tests for the question player

### ISSUE THREAD

**Issue by BenHenning** [oppia/oppia-android#503]
Similar to #388 we need thorough tests for the question player once these can be written in a reliable way for both Espresso & Robolectric.

Test reference: https://github.com/oppia/oppia-android/blob/develop/app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt

**Comment by BenHenning** [comment]
Enough of #89 is now complete that this is no longer blocked.

**Comment by chrk2205** [comment]
@anandwana001  I would like to work on this

**Comment by anandwana001** [comment]
@chrk2205 assigned to you. 
Few points to get start:
1. Write down all the edges case which needs to get tested.
2. Don't include end to end test, example like opening and activity and the  checking any view. Robolectric doesn't support end to end test. 
3. All the test should pass on both espresso and robolectric

How you can send a pr for this:
1. After writing all the points what we need to test, create a comment block in the test file and write the test names and send for review
2. Once above reviewed, you can write test cases. 

Let me know if you need any help in this.

**Comment by chrk2205** [comment]
there are some test cases already present in the `QuestionPlayerActivityTest.kt`
Do I need to write test cases apart from them ??
@anandwana001 
thanks

**Comment by chrk2205** [comment]
@anandwana001 
for test cases should I include this??

- OnBackPressed and its respective cases
- toolbar title display
-

thanks

**Comment by anandwana001** [comment]
Yes, for both comments above

**Comment by ayush0402** [comment]
@anandwana001 Is this issue still needed to be fixed?

**Comment by BenHenning** [comment]
@ayush0402 yes. There's no dedicated test suite for questions currently. We have a few tests in QuestionPlayerActivityTest, but what we really need are QuestionPlayerFragmentLocalTest and QuestionPlayerFragmentTest akin to StateFragmentLocalTest and StateFragmentTest, respectively, that test various interaction & hint scenarios for questions.

**Comment by TanishMoral11** [comment]
Hi @BenHenning , 
Regarding the implementation of `QuestionPlayerFragmentLocalTest` and `QuestionPlayerFragmentTest`, could you please provide further insights into the specific tests that should be included? Understanding the exact scenarios and functionalities to be tested will greatly assist in ensuring comprehensive coverage.

Thanks !!

**Comment by adhiamboperes** [comment]
> Hi [@BenHenning](https://github.com/BenHenning) , Regarding the implementation of `QuestionPlayerFragmentLocalTest` and `QuestionPlayerFragmentTest`, could you please provide further insights into the specific tests that should be included? Understanding the exact scenarios and functionalities to be tested will greatly assist in ensuring comprehensive coverage.
> 
> Thanks !!


@TanishMoral11,

For this type of issues, we generally need to play through features and take notes on layout components and properties, toolbar, when certain UIs appear (such as for hints and solutions), etc. You can share the list with me to review what is important to test/not test.  You will then need to take notes and review what’s currently tested and compare with what should be added (can check with me again on the list before implementing them).

It is otherwise difficult to keep track of what has already been tested and what has not.
